ngraph/frontend/generic/include/frontend_manager/frontend_exceptions.hpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include "ngraph/check.hpp"
+
+namespace ngraph {
+namespace frontend {
+
+enum class FrontEndErrorCode {
+    GENERAL_ERROR,
+    NOT_IMPLEMENTED,
+    OP_VALIDATION_FAILED,
+    NGRAPH_NODE_CREATION_FAILED,
+    INITIALIZATION_ERROR,
+};
```

Comment:
What is the advantage of error codes over specialized error classes? With error code you cannot handle specific types of errors in exception handlers and have to write conditional logic to handle error codes.

Reply:
Fixed